tasks/main.yml

```diff
@@ -38,19 +33,35 @@
 # Amazon Linux (Red Hat) stores similar configuration information in
 # /etc/logrotate.d/syslog, but there the files are already rotated
 # daily.
-- name: Modify rsyslog configuration
+#
+# For Fedora and Debian we have to modify the existing configuration.
+- name: Modify rsyslog configuration (Debian)
   lineinfile:
-    dest: /etc/logrotate.d/rsyslog
+    path: /etc/logrotate.d/rsyslog
     regexp: "{{ item.regex }}"
     state: present
     line: "{{ item.line }}"
     backrefs: yes
   loop:
     # Rotate log files daily
-    - regex: '\s*weekly'
-      line: '\tdaily'
+    - regex: '\s*weekly$'
+      line: "\tdaily"
     # Keep seven days worth of logs
-    - regex: '\s*rotate'
-      line: '\trotate 7'
+    - regex: '\s*rotate \d+$'
```

Comment:
Is there a reason this `regex` and the one above it were not switched from single quotes to double quotes or was this just an oversight?

Reply:
See [here](https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html).  I needed the double quotes for the strings containing escapes (`\t`).  The regexes don't need double quotes since they don't contain escapes; `\s` is a regex thing, not an escape.

Reply:
Yes, that makes sense, but it looks odd because we have a seemingly-inconsistent mix of single-quoting and double-quoting throughout this file (more than just these two regex lines).

Reply:
Well it looks odd, but thus is YAML.  I hesitate to add comments to these lines since I think a decent understanding of YAML's warts should be assumed when working with Ansible.

I will add a comment if you can convince @felddy or @mcdonnnj, though.  I will defer to the wisdom of the herd.

Reply:
I agree with you and don't think that a comment is needed here either.  Although if we are only double-quoting when necessary, then shouldn't these lines be using single quotes?
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L53
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L63
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L65
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L67

Reply:
I agree with @dav3r about consistency. I think double quotes and escaping the backslash (`"\\s*rotate \\d+$"`) is more productive than a comment explaining single vs double quotes if we went that route however.

Reply:
@dav3r and @mcdonnnj, try on commit 30725fe for size.  Let me know what you think.

Reply:
That works for me- thanks and sorry for the extra work!

Reply:
Once @mcdonnnj is happy I will merge.

Reply:
@mcdonnnj confirmed to me that he is happy with the quotes.